src/d3a/models/strategy/external_strategies/load.py

```diff
@@ -437,3 +437,10 @@ def update_energy_forecast(self):
         energy_forecast_Wh = self.energy_forecast_buffer_Wh
         slot_time = self.area.next_market.time_slot
         self.state.set_desired_energy(energy_forecast_Wh, slot_time, overwrite=True)
+        self.state.update_total_demanded_energy(slot_time)
```

Comment:
I think that this is deliberate that this is not set here. Unless you encountered issues with self-sufficiency / self-consumption, I would prefer if you did not add it now. Maybe add a todo or Jira ticket (that will check the kpis in Canary Networks) for future consideration, but better to not change it now IMO.

Reply:
ok. I thought I have found a bug, but did not really think about it. OK, will leave a TODO in the ticket but remove it here.